maps/RandomZLevels/wildwest.dm

```diff
@@ -1,22 +1,392 @@
+
+/obj/item/voucher/free_item/scrip/lazarus
+	name = "lazarus scrip"
+	freebies = list(/obj/item/weapon/lazarus_injector)
+
+/obj/item/voucher/free_item/scrip/rifle
+	name = "rifle scrip"
+	freebies = list(/obj/item/weapon/gun/projectile/hecate/hunting)
+
+/obj/item/voucher/free_item/scrip/sausage
+	name = "sausage scrip"
+	freebies = list(/obj/item/weapon/reagent_containers/food/snacks/sausage)
+
+/obj/item/voucher/free_item/scrip/threefiftyseven
+	name = ".357 scrip"
+	freebies = list(/obj/item/ammo_storage/box/a357)
+
+/obj/machinery/vending/deepvein
+	name = "\improper Deepvein Trust Company Store"
+	desc = "Use your 'wages' here!"
+	product_slogans = list(
+		"Please have your scrip ready for vending."
+	)
+	product_ads = list(
+		"Time is money, so get back to digging!"
+	)
+	vend_reply = "What a glorious time to mine!"
+	icon_state = "mining"
+	vouched = list(
+		/obj/item/weapon/pickaxe/drill = 10,
+		/obj/item/weapon/lazarus_injector = 10,
+		/obj/item/weapon/gun/energy/laser/liberator = 10,
+		/obj/item/weapon/gun/projectile/hecate/hunting = 10,
+		/obj/item/weapon/reagent_containers/food/snacks/sausage = 10,
+		/obj/item/ammo_storage/box/a357 = 20
+		)
+
+
+/obj/item/weapon/card/id/deputy
+	name = "deputy badge"
+	desc = "A metal star that signifies one as a friend of Old Zounds. You're my favorite deputy."
+	assignment = "Deputy"
+	icon_state = "deputystar"
+	access = list(access_deputy)
+	show_biometrics = FALSE
+
+/obj/item/weapon/card/id/deputy/prepickup(mob/user)
+	var/datum/map_element/away_mission/wildwest/W = get_away_mission(WESTERN)
+	if(!W)
+		return 0 //No special away mission
+	if(user in W.wanted)
+		to_chat(user, "<span class='danger'>You spit on \the [src]. A villain would never wear that.</span>")
+		return 1 //Cancel the pickup
+	if(registered_name != "Unknown")
+		return 0 //If someone already registered this badge, don't assign it or send a warning.
+
+	var/decision = alert(user,"Putting on that badge will make people think you're a good guy around Old Zounds, but if you disgrace the badge, you'll be a villain forever.",
+	"Perfidy Warning","I do not shoot with my hand. (Understood)","Get out of Dodge. (Reject badge)")
+	if(decision != "I do not shoot with my hand. (Understood)")
+		return 1
+	if(!Adjacent(user))
+		return 1 //Just a little sanity in case they moved away in that time
+
+	registered_name = user.real_name
+
+	return 0
+
+/obj/structure/uraninitecrystal
+	name = "glowing crystal"
+	icon = 'icons/obj/mining.dmi'
+	icon_state = "crystal"
+	light_color = "#00FF00"
+	anchored = TRUE
+	var/lit = FALSE
+
+/obj/structure/uraninitecrystal/New()
+	..()
+	set_light(2, l_color = light_color)
+
+/obj/structure/uraninitecrystal/bullet_act()
+	rad_pulse()
+
+/obj/structure/uraninitecrystal/kick_act()
+	rad_pulse()
+
+/obj/structure/uraninitecrystal/ex_act()
+	rad_pulse()
+
+/obj/structure/uraninitecrystal/proc/rad_pulse()
+	if(lit)
+		return
+	lit = TRUE
+	set_light(6, l_color = light_color)
+	rad_pulse()
```

Comment:
not sure why this is here because it definitely should not be and causes infinite recursion